src/RouteResolver.tsx

```diff
@@ -51,22 +39,48 @@ export const getRouteFromParams = (pathname: string) => {
       split.push('')
     }
   }
-  let route = routesConfig.find((el) => split[0] === el.to)!
+  let route = ([...sharedRouteConfig, ...routesConfig] as GenericRoute[]).find(
+    (el) => split[0] === el.to,
+  )!
+
+  if (route == null) {
+    console.warn(`Error: url at ${pathname} has no route mapping`)
+    return [homeRoute, '/']
+  }
+
+  let routePathName = '/' + route.to
+
   // search the route configs for the pathname
   for (let i = 1; i < split.length; i += 1) {
-    if (!route) {
-      return fail(`Error: url at ${pathname} has no route mapping`)
-    }
     if (route.isNested) {
-      route = route.routes.find((el) => el.to!.includes(split[i]))!
-    } else {
-      fail(`Route at ${pathname} has no SynapseConfigArray mapping`)
+      const nextRoute = route.routes.find((el) => el.to!.includes(split[i]))
+      if (nextRoute) {
+        route = nextRoute
+        routePathName += '/' + route.to
+      } else {
+        // If we can't find a matching nested route, return the last route that was found
+        break
+      }
     }
   }
-  return route
+
+  // If there's no SynapseConfigArray, then the route we settled on wasn't meant to be a standalone page, and we have nothing to render.
+  // This isn't typical, but just so we can load something, we (recursively) get the first child until we find a route with a SynapseConfigArray
+  while (isEmpty(route.synapseConfigArray) && route.isNested) {
```

Comment:
This part didn't make sense until I looked at the use case described by the test (/Explore/Not-A-Valid-Route -> /Explore/Data).  Nice catch.